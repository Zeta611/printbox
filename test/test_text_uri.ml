let b =
  let open PrintBox in
  vlist
    [
      link ~uri:"#SecondAnchor" @@ frame @@ text "Link to a within-document anchor";
      link ~uri:"https://example.com/1" @@ frame @@ text "child 1";
      link ~uri:"https://example.com/2" @@ text "child 2";
      frame
      @@ tree
           (link ~uri:"https://example.com/3" empty)
           [
             link ~uri:"https://example.com/4"
             @@ tree (frame @@ text "header 3") [ frame @@ text "subchild 3" ];
           ];
      link ~uri:"https://example.com/5"
      @@ tree empty
           [ tree (frame @@ text "header 4") [ frame @@ text "subchild 4" ] ];
      frame @@ text "child 5";
      link ~uri:"https://example.com/6"
      @@ frame
      @@ tree
           (frame @@ text "header 6")
           [
             tree
               (frame @@ text "child 6")
               [
                 link ~uri:"https://example.com/7" @@ frame @@ text "subchild 6";
               ];
           ];
      anchor ~id:"FirstAnchor" @@ frame @@ text "anchor self-link 1";
      tree
        (hlist ~bars:false
           [ anchor ~id:"SecondAnchor" empty; text "silent anchor" ])
        [ text "subchild 7" ];
    ]

let () = print_endline @@ PrintBox_text.to_string b
