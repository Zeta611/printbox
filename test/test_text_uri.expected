Output with ANSI styling:
┌────────────────────────────────┐
│]8;;#SecondAnchor\Link to a within-document anchor]8;;\│
└────────────────────────────────┘
────────────────────────────────────────────────────
┌───────┐
│]8;;https://example.com/1\child 1]8;;\│
└───────┘
────────────────────────────────────────────────────
]8;;https://example.com/2\child 2]8;;\
────────────────────────────────────────────────────
┌──────────────────┐
│──┬────────┐      │
│  │]8;;https://example.com/4\header 3]8;;\│      │
│  ├────────┘      │
│  └─┬──────────┐  │
│    │]8;;https://example.com/4\subchild 3]8;;\│  │
│    └──────────┘  │
└──────────────────┘
────────────────────────────────────────────────────
──┬────────┐
  │]8;;https://example.com/5\header 4]8;;\│
  ├────────┘
  └─┬──────────┐
    │]8;;https://example.com/5\subchild 4]8;;\│
    └──────────┘
────────────────────────────────────────────────────
┌───────┐
│child 5│
└───────┘
────────────────────────────────────────────────────
┌──────────────────┐
│┌────────┐        │
││]8;;https://example.com/6\header 6]8;;\│        │
│├────────┘        │
│└─┬───────┐       │
│  │]8;;https://example.com/6\child 6]8;;\│       │
│  ├───────┘       │
│  └─┬──────────┐  │
│    │]8;;https://example.com/7\subchild 6]8;;\│  │
│    └──────────┘  │
└──────────────────┘
────────────────────────────────────────────────────
{#FirstAnchor}┌──────────────────┐
              │anchor self-link 1│
              └──────────────────┘
────────────────────────────────────────────────────
{#SecondAnchor}silent anchor
└─subchild 7
────────────────────────────────────────────────────
{#ThirdAnchor}anchor self-link 2 after anchor link 2
└─subchild 8
────────────────────────────────────────────────────
]8;;https://example.com/8\external link 8]8;;\ after external link 8
Output without ANSI styling:
uri  │#SecondAnchor
─────┼──────────────────────────────────────────────
inner│┌────────────────────────────────┐
     ││Link to a within-document anchor│
     │└────────────────────────────────┘
─────┼──────────────────────────────────────────────
uri  │https://example.com/1
─────┼──────────────────────────────────────────────
inner│┌───────┐
     ││child 1│
     │└───────┘
─────┼──────────────────────────────────────────────
uri  │https://example.com/2
─────┼──────────────────────────────────────────────
inner│child 2
─────┴──────────────────────────────────────────────
┌─────┬────────────────────────┐
│uri  │https://example.com/3   │
├─────┼─────────────────────   │
│inner│                        │
│└─uri  │https://example.com/4 │
│  ─────┼───────────────────── │
│  inner│┌────────┐            │
│       ││header 3│            │
│       │├────────┘            │
│       │└─┬──────────┐        │
│       │  │subchild 3│        │
│       │  └──────────┘        │
└───────┴──────────────────────┘
─────┬──────────────────────────────────────────────
uri  │https://example.com/5
─────┼──────────────────────────────────────────────
inner│──┬────────┐
     │  │header 4│
     │  ├────────┘
     │  └─┬──────────┐
     │    │subchild 4│
     │    └──────────┘
─────┴──────────────────────────────────────────────
┌───────┐
│child 5│
└───────┘
─────┬──────────────────────────────────────────────
uri  │https://example.com/6
─────┼──────────────────────────────────────────────
inner│┌─────────────────────────────────┐
     ││┌────────┐                       │
     │││header 6│                       │
     ││├────────┘                       │
     ││└─┬───────┐                      │
     ││  │child 6│                      │
     ││  ├──────┬┘                      │
     ││  └─uri  │https://example.com/7  │
     ││    ─────┼─────────────────────  │
     ││    inner│┌──────────┐           │
     ││         ││subchild 6│           │
     ││         │└──────────┘           │
     │└─────────┴───────────────────────┘
─────┴──────────────────────────────────────────────
{#FirstAnchor}┌──────────────────┐
              │anchor self-link 1│
              └──────────────────┘
────────────────────────────────────────────────────
{#SecondAnchor}silent anchor
└─subchild 7
────────────────────────────────────────────────────
{#ThirdAnchor}anchor self-link 2 after anchor link 2
└─subchild 8
─────┬──────────────────────────────────────────────
uri  │https://example.com/8 after external link 8
─────┼─────────────────────
inner│external link 8
